scripts/updateNpm.js

```diff
@@ -0,0 +1,47 @@
+// @ts-check
+
+import { writeFileSync } from 'node:fs';
+import { promisify } from 'node:util';
+import { exec } from 'node:child_process';
+import { componentsSchema } from '../src/lib/schemas.js';
+import components from '../src/routes/components/components.json' assert { type: 'json' };
+
+const execAsync = promisify(exec);
+
+const data = componentsSchema.parse(components);
+
+const { versions, dates, support } = await Promise.all(
+	data.map(async (pkg) => {
+		try {
+			return await processPackage(pkg);
+		} catch (error) {
+			console.log(error.message);
+		}
+	})
+).then((values) => {
+	let versions = {};
+	let dates = {};
+	let support = {};
+	for (const value of values) {
+		if (value) {
+			versions[value.name] = value.version;
+			dates[value.name] = value.date;
+			support[value.name] = value.support;
+		}
+	}
+	return { versions, dates, support };
+});
```

Comment:
This doesn't add anything, and it's probably worse in terms of performance and memory usage.

It can be easier to read of some, but your version is pretty logical (a `for` loop, a `if` and 3 variables is way more standard than a `.filter(Boolean)` and a `.reduce`)

My version duplicate the data 2 times in memory (one in the `.filter`, one in the `.reduce`) where you version only once (result of the `for` loop)  
_My version won't probably never by optimized by any compiler_

```suggestion
).then((values) => values
	.filter(Boolean)
	.reduce((result, value) => {
		result.versions[value.name] = value.version;
		result.dates[value.name] = value.date;
		result.support[value.name] = value.support;
		return result;
	}, {versions: {}, dates: {}, support: {}})
);
```